spec/InvertedIndex.spec.js

```diff
@@ -0,0 +1,292 @@
+   });
+
+   it('should return an object with searched tokens' +
+     ' and their indexes when it is invoked ',
+    () => {
+      expect(invertedIndex.searchIndex(tokens, indexx)).toEqual({
+        hello: [true, true, false],
+        world: [false, false, false],
+        alice: [false, false, false],
+        man: [true, false, false]
+      });
+    });
+ });
+
+describe('updateSearchResult method',
+ () => {
+   let tokens;
+   let file;
+   beforeEach(() => {
+     tokens = ['alice', 'ee', 'enters'];
+     file = 'touch.json';
+   });
+
+   it('should return "object" when its type is checked', () => {
+     expect(typeof (invertedIndex.updateSearchResult(file, tokens)))
+     .toBe('object');
+   });
+
+   it('should update the search record when it is invoked', () => {
+     expect(invertedIndex.updateSearchResult(file, tokens)).toEqual({
+       'touch.json': {
+         alice: [true, false, false],
+         ee: [false, false, true],
+         enters: [true, false, false]
+       }
+     });
+   });
+ });
+
+
+describe('helpers.fileIsValid function',
+ () => {
+   let file1;
+   let file2;
+   let file3;
+   let file4;
+   beforeEach(() => {
+     file1 = { name: 'book.json' };
+     file2 = { name: 'book.css' };
+     file3 = { name: 'book' };
+     file4 = { name: 'book.JSON' };
+   });
+
+   it('should return "true" for a valid json file when its is invoked', () => {
+     expect(helpers.fileIsValid(file1)).toBe(true);
+     expect(helpers.fileIsValid(file4)).toBe(true);
+   });
+
+   it('should return "false" for an invalid file when it is invoked', () => {
+     expect(helpers.fileIsValid(file2)).toBe(false);
+     expect(helpers.fileIsValid(file3)).toBe(false);
+   });
+ });
+
+describe('helper.validateFileContents function',
+ () => {
+   let book1;
+   let book2;
+   let book3;
+   let book4;
+   beforeEach(() => {
+     book1 = [{ title: 'Alice , / ?', text: 'enters a a.' },
+     { title: 'Fellowship )&* ...', text: 'wizard on on' },
+     { title: 'Thee + = - ee', text: 'un un usuals' }];
+     book2 = { men: 'kskkskskskkskskskksksk' };
+     book3 = { boy: 'usman', girl: 'dammy' };
+     book4 = 'this is the way we roll around here';
+   });
+
+   it('should return boolean when invoked with' +
+   ' a valid json file input', () => {
+     expect(typeof (helpers.validFileContent(book1))).toBe('boolean');
+   });
+
+   it('should return true when invoked with' +
+   ' a valid file contents', () => {
+     expect(helpers.validFileContent(book1)).toBe(true);
+   });
+
+   it('should return false when invoked with' +
+   ' an invalid data format', () => {
+     expect(helpers.validFileContent(book2)).toBe(false);
+     expect(helpers.validFileContent(book3)).toBe(false);
+     expect(helpers.validFileContent(book4)).toBe(false);
+   });
+   it('should return false when invoked with' +
+   ' an empty book', () => {
+     expect(helpers.validFileContent([{}])).toBe(false);
```

Comment:
There are ways to write the specs without needing line breaks. Also it's not descriptive enough.
For example `should return boolean when invoked with +   ' a valid json file input` could be written as `should return true when passed a valid json file`